tests/Neo.UnitTests/SmartContract/Manifest/UT_ExtendedType.cs

```diff
@@ -0,0 +1,276 @@
+            Assert.IsTrue(json["forbidnull"]?.AsBoolean());
+            Assert.AreEqual("IIterator", json["interface"]?.AsString());
+            Assert.AreEqual("String", json["key"]?.AsString());
+        }
+
+        [TestMethod]
+        public void FromJson_ToJson_Roundtrip()
+        {
+            var json = new JObject();
+            json["type"] = "Boolean";
+            json["namedtype"] = "Type.A";
+            json["length"] = 32;
+            json["forbidnull"] = false;
+            json["interface"] = "IIterator";
+            json["key"] = "PublicKey";
+
+            var ext = ExtendedType.FromJson(json);
+            var output = ext.ToJson();
+
+            Assert.AreEqual("Boolean", output["type"]?.AsString());
+            Assert.AreEqual("Type.A", output["namedtype"]?.AsString());
+            Assert.AreEqual(32, output["length"]?.AsNumber());
+            Assert.IsFalse(output["forbidnull"]?.AsBoolean());
+            Assert.AreEqual("IIterator", output["interface"]?.AsString());
+            Assert.AreEqual("PublicKey", output["key"]?.AsString());
+        }
+
+        [TestMethod]
+        public void FromJson_InvalidType_ShouldThrow()
+        {
+            var json = new JObject();
+            json["type"] = "InvalidType";
+            Assert.ThrowsExactly<ArgumentException>(() => ExtendedType.FromJson(json));
+        }
+
+        [TestMethod]
+        public void FromJson_NegativeLength_ShouldThrow()
+        {
+            var json = new JObject();
+            json["type"] = "ByteArray";
+            json["length"] = -1;
+            Assert.ThrowsExactly<FormatException>(() => ExtendedType.FromJson(json));
+        }
+
+        [TestMethod]
+        public void FromJson_InvalidInterface_ShouldThrow()
+        {
+            var json = new JObject();
+            json["type"] = "InteropInterface";
+            json["interface"] = "BadInterface";
+            Assert.ThrowsExactly<FormatException>(() => ExtendedType.FromJson(json));
+        }
+
+        [TestMethod]
+        public void FromJson_InvalidKey_ShouldThrow()
+        {
+            var json = new JObject();
+            json["type"] = "Map";
+            json["key"] = "BadKey";
+            Assert.ThrowsExactly<FormatException>(() => ExtendedType.FromJson(json));
+        }
+
+        [TestMethod]
+        public void ToStackItem_And_FromStackItem_ShouldRoundtrip()
+        {
+            var original = new ExtendedType
+            {
+                Type = ContractParameterType.Map,
+                NamedType = "MapType",
+                Length = 20,
+                ForbidNull = true,
+                Interface = Nep25Interface.IIterator,
+                Key = ContractParameterType.Hash160
+            };
+
+            var refCounter = new ReferenceCounter();
+            var structItem = original.ToStackItem(refCounter);
+
+            var copy = new ExtendedType();
+            copy.FromStackItem((VM.Types.Array)structItem);
```

Comment:
```suggestion
            copy.FromStackItem(structItem);
```
Fails to cast to an array